### ISSUE TITLE
Can the Online Editor Include Access for Running the Code without Submitting?

### ISSUE THREAD

**Issue by thelmalu** [exercism/exercism#5873]
I'm working on the Ruby Track and I run a lot of code before I get near something complete enough to run the tests on. I've tried to work with the online editor, but this lack has forced me back onto my own machine.

**Comment by angelikatyborska** [comment]
I am not sure what you mean. You should be able to run the tests in the web editor as many times as you need without submitting the solution. Maybe you could take a screenshot of the problem?

**Comment by thelmalu** [comment]
When I'm working through a problem, I often want to test fragments of code: some might not even look like they have anything to do with solving the problem--I might not even have a class or module defined at that point.  I want to insert test prints, to see the long tracebacks that the Ruby interpreter shows me, not just 'ERROR' with a minimal idea of what it is, or 'FAILED' with no further info

**Comment by joshgoebel** [comment]
Online REPLs, fun. :)

**Comment by iHiD** [comment]
You can use `debug` to print things. I think that's what you mean. As Angelika says, you can run the test without submitting the solution.

**Comment by thelmalu** [comment]
As I said, I often want to test fragments of code before they're at all recognizably connected to the current problem--hmm, maybe I can fake this by commenting out every one of the tests, but that would still leave an unwanted testing framework

**Comment by NobbZ** [comment]
That is not what the online editor is meant for. It is meant as a very basic tool to do a solution to exactly this exercise in exactly the skeleton given. If you want to do some REPL stuff or carve out your code by different means, then using the CLI to download the exercise and doing local editing/carving is your best bet.

**Comment by thelmalu** [comment]
Yes, Norbert Melzer.  I will have to continue with the command line interface

**Comment by joshgoebel** [comment]
Many languages have online REPLs also... might be worth a google... but that isn't a thing that Exercism provides.
